ashlar/filters.py

```diff
@@ -84,6 +87,66 @@ class Meta:
         fields = ['data', 'boundary']
 
 
+class DateRangeFilterBackend(BaseFilterBackend):
+    """Used to filter querysets based on a given START_FIELD and END_FIELD
+
+    NOTE: This filter must be inherited from in order to be used because
+    1. its Meta class must be set with a model and list of fields
+    2. you likely want to override START_FIELD/END_FIELD
+
+    How this filter works:
+    If any portion of the 'valid' datetime ranges overlap, the object is returned;
+    however, the 'end time' is not inclusive, so if the 'valid start period datetime'
+    is equal to the query value 'valid_to' that mission will be excluded.
+
+    If only 'valid_from' or 'valid_to' is provided this operates similar to > or <
+    e.g. 'valid_from=1901-01-01T00:00:00+00:00' will return all missions where the mission's
+    valid period is greater than 1901-01-01T00:00:00+00:00.
+
+    If both 'valid_from' and 'valid_to' are the same datetime then missions that include
+    that datetime in its valid period are returned.
+
+    For greater explanation, see the diagrams/tests for MissionFilter in missions/tests.py
+    """
+
+    MIN_DATETIME = '1901-01-01T00:00:00+00:00'
+    MAX_DATETIME = '9999-12-31T23:59:59.999999+00:00'
+
+    FIELD = 'occurred_from'
+
+    def filter_queryset(self, request, queryset, view):
+        """Filter records by date
+
+        Arguments
+        :param request: django rest framework request instance
+        :param queryset: queryset to apply filter to
+        :param view: view that this filter is being used by
+
+        QUERY PARAMS
+        :param valid_from: ISO 8601 timestamp
+        :param valid_to: ISO 8601 timestamp
```

Comment:
Looks like these and the big comment above need to be updated to match this functionality.